### PR TITLE
Button / TransactionButton component style updates

### PR DESCRIPTION
**Pull request by am9u**
- Add DarkButton
- Add Inverted and Dark TransactionButton styles
- Add support for custom pre/post transtions AOP methods to
TransactionButton

### CODE REVIEW

File: packages/components/src/TransactionButton.tsx [hunk 1]
```diff
@@ -72,6 +73,37 @@ export class TransactionButtonNoModal extends React.Component<TransactionButtonP
   }
 
   public render(): JSX.Element {
+    if (this.props.style && this.props.style === "inverted") {
```
Comment (walfly) [inline packages/components/src/TransactionButton.tsx]: should this be a switch? or assign a variable like `Button` to the button component you want.

```
let Button: StyledComponentClass;
if (whatever condition) {
  Button = DarkButton;
}
return <Buton {...props}/>;
```

File: packages/components/src/TransactionButton.tsx [hunk 2]
```diff
@@ -72,14 +73,29 @@ export class TransactionButtonNoModal extends React.Component<TransactionButtonP
   }
 
   public render(): JSX.Element {
+    let StyledButton;
```
Comment (am9u) [inline packages/components/src/TransactionButton.tsx]: Ok, I ended up taking both of your suggestions @walfly.  Thx, it def looks more readable.

File: packages/components/src/Button.tsx [hunk 3]
```diff
@@ -113,6 +113,22 @@ export const SecondaryButton = BaseButton.extend`
...
+  &.active {
+    background-color: ${colors.accent.CIVIL_GRAY_1};
+    border: 1px solid ${colors.accent.CIVIL_GRAY_1};
+    color: ${colors.basic.WHITE};
+  }
+  &.active {
+    background-color: ${colors.accent.CIVIL_GRAY_1};
+    border: 1px solid ${colors.accent.CIVIL_GRAY_1};
+    color: ${colors.basic.WHITE};
+  }
```
Comment (sruddy) [inline packages/components/src/Button.tsx]: Should the second `&.active` be a different state? I see it happens on `SecondaryButton` also.

Comment (am9u) [reply packages/components/src/Button.tsx]: Ah, good catch. I was copy-pasta-ing those definitions from `SecondaryButton`, but my guess is that they're supposed to be `:hover` definitions in there. I'm updating with that assumption, and we can address further if the design team has feedback.

**Pull request merged**